docs/architecture/clients/services/implementation.md

```diff
@@ -0,0 +1,200 @@
+---
+sidebar_position: 2
+---
+
+# Implementation
+
+Bitwarden's client service architecture is aligned with the [vision](vision.md) that we have
+defined. These goals were the driving force behind the current implementation.
+
+## Different Types of Services
+
+The Bitwarden clients have a service architecture that is comprised of primarily three different
+types of services, for a given domain in our [Data Model](../data-model.md):
+
+- `[Domain]Service`
+- `Internal[Domain]Service`
+- `[Domain]ApiService`
+
+```kroki type=plantuml
+@startuml
+abstract class Internal[Domain]Service #aliceblue {
+  + upsert(item: [Domain])
+  + replace(items: [Domain][])
+}
+class [Domain]Service #aliceblue {
+  # subject : BehaviorSubject<[Domain][]>
+  + observable: Observable<[Domain][]>
+  - updateObservables()
+}
+class [Domain]ApiService #aliceblue {
+  + get[Domain]()
+  + get[Domain]by[Property]()
+  + put[Domain]()
+}
+class "UI Component"
+class SyncService {
+  + fullSync()
+  - sync[Domain]()
+}
+class ApiService {
+  + send()
+}
+
+"UI Component" *-- "[Domain]Service" : uses
+"UI Component" *-- "[Domain]ApiService" : uses
+
+"[Domain]Service" --|> "Internal[Domain]Service" : extends
+SyncService *-- "Internal[Domain]Service" : uses
+
+"[Domain]ApiService" *-- ApiService : uses
+"[Domain]ApiService" *-- "Internal[Domain]Service" : uses
+@enduml
+```
+
+The classes and their responsibilities are detailed below:
+
+| Type                      | Responsibility                                                                                                                                                                                                                                | Example                 |
+| ------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------- |
+| `[Domain]Service`         | Provides read access to the domain through an Observable, which is updated when the local state changes via an RxJS `BehaviorSubject`. This service also provides helper methods related to that domain.                                      | `PolicyService`         |
+| `Internal[Domain]Service` | Provides public update methods on the service's Observable properties _without_ updating the `[Domain]` on the server. It is implemented as an abstract class which is extends `[Domain]Service`, but generally not shared with most classes. | `InternalPolicyService` |
+| `[Domain]ApiService`      | Provides synced-with-server write capabilities. In general, if you’re editing a `[Domain]`, you want `[Domain]ApiService`.                                                                                                                    | `PolicyApiService`      |
+
+:::tip Why a separate ApiService for the domain?
```

Comment:
This title gets capitalized, which makes it hard to read `ApiService`. I suggest either wrapping as inline code (`ApiService`) or two words (API Service)

Reply:
I removed this section.